src/vanilla/Model/CompositeTypeCs.cs

```diff
@@ -159,13 +159,29 @@ public string Signature
                     foreach (PropertyCs property in Parameters.Where(p => !p.UnderlyingProperty.IsConstant).Select(p => p.UnderlyingProperty))
                     {
                         string format = (property.IsRequired ? "{0} {1}" : "{0} {1} = default({0})");
-                        // for people who really want defaults to properties (be aware of the PATCH operation consequences!)
-                        if (property.UseDefaultInConstructor && property.DefaultValue != null)
+                        
+                        // we need some special signatures for extensible enums
+                        if(property.ModelType is EnumType && Settings.Instance.CustomSettings.ContainsKey("ExtensibleEnums") && (bool)Settings.Instance.CustomSettings["ExtensibleEnums"])
                         {
-                            format = "{0} {1} = " + property.DefaultValue;
+                            // for people who really want defaults to properties (be aware of the PATCH operation consequences!)
+                            var enumProp = property.ModelType as EnumType;
+                            if(property.UseDefaultInConstructor && enumProp.DefaultValue!=null)
+                            {
+                                format = "{0} {1} = " + enumProp.DefaultValue;

```

Comment:
why is this a special case? I could very well be missing something right now, but why would we take DefaultValue out of the enum *type* instead of the property?

Reply:
This is slightly different `Property.ModelType.DefaultValue`

Reply:
How? My point is: default values are something you specify on *usage* of a type (property, parameter), not the type itself - so why resort to the type here? In essence, it's a weird glitch in our current C# types that types even *have* a `DefaultValue`. In other words: If I have a Swagger with some extensible enum and say `default: 3`, are you sure that `enumProp.DefaultValue` will contain 3?

Reply:
I remember now, the thing is, by default, we set the type for the default property as `property.ModelTypeName` but since we are creating a new type for the enum, we need `enumProp.ClassName`.

Reply:
aha! but shouldn't that happen at the level of `property.DefaultValue`? Like this, this is very problematic:
What will `(property.ModelType as EnumType).DefaultValue` contain if one and the same enum is used twice, and with different default values? It simply *can't* be the correct value in both cases since `property.ModelType as EnumType` will point to the same object. *Only* `property.DefaultValue` has the fidelity to distinguish both usages. This is similar to the type definition vs. type usage issues we once had with `required` 🙂

Reply:
Ahh okay, I see, I will use the defaultValue from property then, ( I still need the typecasting to enumtype for reasons mentioned above)